File: day:55/levelOrder.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    if(root === null) return [];
    let result = [];
    let stack = [root];
    
    while(stack.length > 0) {
        let newStack = [];
        let temp = [];
        while(stack.length > 0) {
            let node = stack.shift();
            temp.push(node.val);
            if(node.left) {
                newStack.push(node.left);
            }
            if(node.right) {
                newStack.push(node.right);
            }
        }
        stack = newStack;
        result.push(temp);
    }
    return result;
    
};

File: day:55/levelOrder.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if(root is None): return [];
        result = [];
        stack = [root];
    
        while(len(stack) > 0):
            newStack = [];
            temp = [];
            while(len(stack) > 0):
                node = stack.pop(0);
                temp.append(node.val)
                if(node.left):
                 newStack.append(node.left)
                if(node.right):
                    newStack.append(node.right)
            stack = newStack;
            result.append(temp);
        return result
